src/MPAGSCipher/CaesarCipher.cpp

```diff
@@ -1,38 +1,49 @@
-#include "RunCaesarCipher.hpp"
+#include "CaesarCipher.hpp"
+#include "CipherMode.hpp"
 
+#include <iostream>
 #include <string>
-#include <vector>
 
-std::string runCaesarCipher(const std::string& inputText, const std::size_t key,
-                            const bool encrypt)
-{
+CaesarCipher::CaesarCipher(const std::size_t cipherKey) {
+    key_ = cipherKey;
+}
+CaesarCipher::CaesarCipher(const std::string cipherKey) {
+    if (cipherKey.empty()) {
+        key_ = 0;
+    }
+    else {
+        key_ = std::stoul(cipherKey);
```

Comment:
Was expecting to see the code that's currently in lines 90-112 of mpegs-cipher.cpp to be moved into here.  It's the responsibility of this class to check that the provided string can be used ok.

As above, can do the truncation of the key here:
```suggestion
        key_ = std::stoul(cipherKey) % alphabet_.size();
```